cvmfs/network/sink_file.cc

```diff
@@ -41,6 +41,23 @@ int FileSink::Reset() {
           (freopen(NULL, "w", file_) == file_)) ? 0 : -errno;
 }
 
+/**
+ * Purges all resources leaving the sink in an invalid state.
+ * More aggressive version of Reset().
+ * For some sinks and depending on owner status it might do 
+ * the same as Reset().
+ *
+ * @returns Success = 0
+ *          Failure = -errno
+ */
+int FileSink::Purge() {
+  if (is_owner_ && file_) {
+    return (-1 * fclose(file_));
```

Comment:
You need to check `fclose` for `EOF` return value and then return `-errno` from this method.

Reply:
Also, set `file_` to `NULL` after the `fclose()` call.